lib/secretsManager.js

```diff
@@ -1,77 +1,173 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import uuid from 'uuid-random'
+import {
+  encryptWithGeneratedKey,
+  encryptWithKeyDerivedFromString,
+  decryptWithKeyDerivedFromString,
+  utf8ToBytes,
+  bytesToUtf8,
+  CipherStrategy,
+} from '@meeco/cryppo'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  static singleton = null
+
+  static async getInstance() {
+    if (!SecretsManager.singleton) {
+      SecretsManager.singleton = new SecretsManager()
+      await SecretsManager.singleton.init()
+    }
+
+    return SecretsManager.singleton
+  }
+
+  constructor() {
+    this.encryptionKey = null
+  }
+
+  async init() {
+    if (SecureStore.isAvailableAsync()) {
+      await this.initializeWithSecureRandomPassphrase()
+    } else {
+      await this.initializeWithVulnerablePassphrase()
+    }
+  }
+
+  async initializeWithSecureRandomPassphrase() {
+    const encryptionKey = await SecureStore.getItemAsync(
+      encryptionPassphraseStorageKey
     )
-  } catch (err) {
-    console.error('Failed to read secrets from data store', err)
+
+    if (encryptionKey) {
+      this.encryptionKey = encryptionKey
+    } else {
```

Comment:
minor: you can easily avoid if/else if you return early. makes the code less nested and easier to read

Reply:
Yup! I can change that too.